prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1116 @@
+                    Pair(
+                        it,
+                        BlobAccess.downloadBlobAsByteArray(it.bodyUrl, getBlobContainerMetadata())
+                    )
+                }
+
+            assertThat(reportAndBundles).transform { pairs ->
+                pairs.map {
+                    it.second.toString(Charsets.UTF_8)
+                }
+            }.containsOnly(validFHIRRecord1)
+
+            // is the queue messaging what we expect it to be?
+            val expectedRouteQueueMessages = reportAndBundles.flatMap { (report, fhirBundle) ->
+                listOf(
+                    FhirTranslateQueueMessage(
+                        report.reportId,
+                        report.bodyUrl,
+                        BlobAccess.digestToString(BlobAccess.sha256Digest(fhirBundle)),
+                        "phd.fhir-elr-no-transform",
+                        fhirSenderWithNoTransform.topic,
+                        "phd.x"
+                    ),
+                    FhirTranslateQueueMessage(
+                        report.reportId,
+                        report.bodyUrl,
+                        BlobAccess.digestToString(BlobAccess.sha256Digest(fhirBundle)),
+                        "phd.fhir-elr-no-transform",
+                        fhirSenderWithNoTransform.topic,
+                        "phd.y"
+                    )
+                )
+            }.map {
+                it.serialize()
+            }
+
+            verify(exactly = 2) {
+                QueueAccess.sendMessage(
+                    elrTranslationQueueName,
+                    match {
+                        expectedRouteQueueMessages.contains(it)
+                    }
+                )
+            }
+
+            // make sure action table has a new entry
+            checkActionTable(listOf(TaskAction.convert, TaskAction.receive, TaskAction.route))
+
+            // ACTION_LOG table is expected to have a record with no record_id identifying an observation filter action
+            // that may or may not have happened.
+            checkActionLogTable(expectedButStrangeObservationFilterEntry)
+        }
+    }
+
+    //@Ignore
+    @Test
+    fun `should send valid FHIR report filtered by condition code 94558-5`() {
+        // set up
+        val reportContents =
+            listOf(
+                File(MULTIPLE_OBSERVATIONS_FHIR_URL).readText()
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val receiveReport = reportPair.first
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        var receiverSetupData = listOf(
```

Comment:
Last time I'll note this, but IntelliJ suggests using val instead. Makes sure you go through IDE suggestions and accept them if there is not a good reason to not.